study1/week2/Range_Sum_of_BST.java

```diff
@@ -0,0 +1,28 @@
+class Range_Sum_of_BST {
+    int sum;
+
+    public int rangeSumBST(TreeNode root, int low, int high) {
+
+        if (low <= root.val && root.val <= high) {
+            sum += root.val;
+        }
+
+        if (root.left != null) {
+            rangeSumBST(root.left, low, high);
+        }
+
+        if (root.right != null) {
+            rangeSumBST(root.right, low, high);
+        }
+
+        return sum;
+    }
+    /**
+     * Time Complexity : O(n)
+       The algorithm visits each node in the binary tree once, resulting in a linear time complexity.
+
+     * Space Compexity : O(1)
```

Comment:
So, you are almost right, except you also need to consider the size of the stack being utlized. 
As you said, this algorithm visits every node at least once in the worst case. So if the BST is unbalanced, you will have n nodes in the call stack, which means that this will consume `O(n)` space as well.